src/Smart.Design.Razor/TagHelpers/Header/HeaderHtmlGenerator.cs

```diff
@@ -0,0 +1,192 @@
+
+        return divLeft1;
+    }
+
+    private static IHtmlContent GetDivRight1(string currentLanguage, Dictionary<string, string> languagesAndLink)
+    {
+        var divRight1 = new TagBuilder("div");
+        divRight1.AddCssClass("c-toolbar__item");
+
+        var nav = new TagBuilder("nav");
+        var languageList = new TagBuilder("ul");
+        languageList.AddCssClass("c-pill-navigation");
+
+        var languageLi = new TagBuilder("li");
+        languageLi.AddCssClass("c-pill-navigation__item c-pill-navigation__item--has-child-menu");
+
+        var languageButton = new TagBuilder("button");
+        languageButton.Attributes["type"] = "button";
+        languageButton.Attributes["data-menu"] = "lang-switch";
+        languageButton.InnerHtml.Append(currentLanguage);
+
+        var languageUl = new TagBuilder("ul");
+        languageUl.AddCssClass("c-menu c-menu--large");
+        languageUl.Attributes["id"] = "lang-switch";
+
+        foreach (KeyValuePair<string, string> item in languagesAndLink)
+        {
+            var li = new TagBuilder("li");
+            li.AddCssClass("c-menu__item");
+
+            var href = new TagBuilder("a");
+            href.AddCssClass("c-menu__label");
+            href.Attributes["href"] = item.Value;
+            href.InnerHtml.Append(item.Key);
+
+            li.InnerHtml.AppendHtml(href);
+            languageUl.InnerHtml.AppendHtml(li);
+        }
+
+        languageLi.InnerHtml.AppendHtml(languageButton);
+        languageLi.InnerHtml.AppendHtml(languageUl);
+
+        languageList.InnerHtml.AppendHtml(languageLi);
+        nav.InnerHtml.AppendHtml(languageList);
+        divRight1.InnerHtml.AppendHtml(nav);
+
+        return divRight1;
+    }
+
+    private static IHtmlContent GetDivRight2(string userName, string avatarPath, Dictionary<string, string> labelsAndLinks)
+    {
+        var divRight2 = new TagBuilder("div");
+        divRight2.AddCssClass("c-toolbar__item");
+
+        var button = new TagBuilder("button");
+        button.AddCssClass("c-user-navigation");
+        button.Attributes["data-menu"] = "userMenu";
+        button.Attributes["href"] ="#";
+
+        var divImage = new TagBuilder("div");
+        divImage.AddCssClass("c-avatar c-avatar--img");
+
+        var image = new TagBuilder("img");
+        image.Attributes["src"] = avatarPath;
+        image.Attributes["alt"] = "avatar";
+
+        divImage.InnerHtml.AppendHtml(image);
+
+        var fullName = new TagBuilder("p");
+        fullName.AddCssClass("c-avatar-and-text__text");
+        fullName.InnerHtml.Append(userName);
+
+        button.InnerHtml.AppendHtml(divImage);
+        button.InnerHtml.AppendHtml(fullName);
+
+        var menu = new TagBuilder("ul");
+        menu.AddCssClass("c-menu c-menu--large");
+        menu.Attributes["id"] = "userMenu";
+
+        var menuInfo = new TagBuilder("li");
+        menuInfo.AddCssClass("c - menu__info");
+
+        var divAvaterText = new TagBuilder("div");
```

Comment:
```suggestion
        var divAvatarText = new TagBuilder("div");
```
Small typo, just to be thorough